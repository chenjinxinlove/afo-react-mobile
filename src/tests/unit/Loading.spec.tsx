import { shallow } from 'enzyme';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Loading from '../../pageages/loading/';
import DemoLoading from '../../pageages/loading/demo';

describe('Button', () => {
  // 测试props
  it('props', () => {
    const wrapper = shallow(<Loading
      color = 'white'
      size = {40}
    />)
    expect(wrapper.find('.afo-loading-spinners').prop('style')).toEqual({"width": "40px", "height": "40px"})
  })
  // 测试快照
  it('test snapshots', () => {
    const component = renderer.create(
      <DemoLoading />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
 