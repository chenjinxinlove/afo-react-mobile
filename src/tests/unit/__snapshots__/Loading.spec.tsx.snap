// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test snapshots 1`] = `
<div>
  <div
    className="afo-loading afo-loading-black"
  >
    <span
      className="afo-loading-spinners"
      style={
        Object {
          "height": "30px",
          "width": "30px",
        }
      }
    >
      <i
        className="afo-loading-spinner"
      />
      <i
        className="afo-loading-spinner"
      />
      <i
        className="afo-loading-spinner"
      />
      <i
        className="afo-loading-spinner"
      />
      <i
        className="afo-loading-spinner"
      />
      <i
        className="afo-loading-spinner"
      />
      <i
        className="afo-loading-spinner"
      />
      <i
        className="afo-loading-spinner"
      />
      <i
        className="afo-loading-spinner"
      />
      <i
        className="afo-loading-spinner"
      />
      <i
        className="afo-loading-spinner"
      />
      <i
        className="afo-loading-spinner"
      />
    </span>
  </div>
  <div
    className="afo-loading afo-loading-white"
  >
    <span
      className="afo-loading-spinners"
      style={
        Object {
          "height": "30px",
          "width": "30px",
        }
      }
    >
      <i
        className="afo-loading-spinner"
      />
      <i
        className="afo-loading-spinner"
      />
      <i
        className="afo-loading-spinner"
      />
      <i
        className="afo-loading-spinner"
      />
      <i
        className="afo-loading-spinner"
      />
      <i
        className="afo-loading-spinner"
      />
      <i
        className="afo-loading-spinner"
      />
      <i
        className="afo-loading-spinner"
      />
      <i
        className="afo-loading-spinner"
      />
      <i
        className="afo-loading-spinner"
      />
      <i
        className="afo-loading-spinner"
      />
      <i
        className="afo-loading-spinner"
      />
    </span>
  </div>
</div>
`;
